opencog/atomspace/TruthValue.cc

```diff
@@ -94,3 +94,13 @@ bool TruthValue::isDefaultTV() const
     }
     return false;
 }
+
+TruthValueType TruthValue::getType() const
+{
+	return tv_type_;
+}
+
+void TruthValue::setType(TruthValueType t)
+{
+	tv_type_ = t;
```

Comment:
Hmm, I don't have any opposition about the optimization issue, so this PR might be turned down for that reason.If I have understood you correctly, you are in favor of changing the signature of the merge method (besides the naming issue) and oppose every other changes introduced. Right?